test/e2e/kamsuite/kamsuite.go

```diff
@@ -95,47 +97,83 @@ func FeatureContext(s *godog.Suite) {
-			if !ok {
-				fmt.Println(errMessage)
-			}
-		default:
-			fmt.Println("SCM is not supported")
+		err := deleteRepository(os.Getenv("GITOPS_REPO_URL"))
+		if err != nil {
+			fmt.Println(os.Getenv("GITOPS_REPO_URL"), ":", err)
 		}
 	})
 }
 
+type VCS struct {
+	GITHUB GIT `json:"github"`
+	GITLAB GIT `json:"gitlab"`
+}
+
+type GIT struct {
+	SERVICE_REPO_URL      string `json:"service_repo_url"`
+	GITOPS_REPO_URL       string `json:"gitops_repo_url"`
+	IMAGE_REPO            string `json:"image_repo"`
+	BUS_REPO_URL          string `json:"bus_repo_url"`
+	DOCKERCONFIGJSON_PATH string `json:"dockerconfigjson_path"`
+	GIT_ACCESS_TOKEN      string `json:"git_access_token"`
+}
+
 func envVariableCheck() bool {
-	envVars := []string{"SERVICE_REPO_URL", "GITOPS_REPO_URL", "IMAGE_REPO", "DOCKERCONFIGJSON_PATH", "GIT_ACCESS_TOKEN", "BUS_REPO_URL"}
+
+	var stdout bytes.Buffer
+	var vcs VCS
+
 	val, ok := os.LookupEnv("CI")
 	if !ok {
+
+		cmd := exec.Command("pwd")
+		cmd.Stdout = &stdout
+
+		err := cmd.Run()
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		jsonFile, err := os.Open(strings.Split(stdout.String(), "/test")[0] + "/variables.json")
+		if err != nil {
+			fmt.Println(err)
+			fmt.Println("Configuration file not set")
+			return false
+		}
+
+		defer jsonFile.Close()
+
+		byteValue, _ := ioutil.ReadAll(jsonFile)
+		json.Unmarshal(byteValue, &vcs)
+
+		if os.Getenv("VCS") == "github" {
+			os.Setenv("SERVICE_REPO_URL", vcs.GITHUB.SERVICE_REPO_URL)
+			os.Setenv("GITOPS_REPO_URL", vcs.GITHUB.GITOPS_REPO_URL)
+			os.Setenv("IMAGE_REPO", vcs.GITHUB.IMAGE_REPO)
+			os.Setenv("BUS_REPO_URL", vcs.GITHUB.BUS_REPO_URL)
+			os.Setenv("DOCKERCONFIGJSON_PATH", vcs.GITHUB.DOCKERCONFIGJSON_PATH)
+			os.Setenv("GIT_ACCESS_TOKEN", vcs.GITHUB.GIT_ACCESS_TOKEN)
+		} else if os.Getenv("VCS") == "gitlab" {
+			os.Setenv("SERVICE_REPO_URL", vcs.GITLAB.SERVICE_REPO_URL)
+			os.Setenv("GITOPS_REPO_URL", vcs.GITLAB.GITOPS_REPO_URL)
+			os.Setenv("IMAGE_REPO", vcs.GITLAB.IMAGE_REPO)
+			os.Setenv("BUS_REPO_URL", vcs.GITLAB.BUS_REPO_URL)
+			os.Setenv("DOCKERCONFIGJSON_PATH", vcs.GITLAB.DOCKERCONFIGJSON_PATH)
+			os.Setenv("GIT_ACCESS_TOKEN", vcs.GITLAB.GIT_ACCESS_TOKEN)
+		} else {
+			fmt.Println("VCS not set")
+			os.Exit(1)
+		}
+
+		envVars := []string{"SERVICE_REPO_URL", "GITOPS_REPO_URL", "IMAGE_REPO", "DOCKERCONFIGJSON_PATH", "GIT_ACCESS_TOKEN", "BUS_REPO_URL"}
 		for _, envVar := range envVars {
-			_, ok := os.LookupEnv(envVar)
-			if !ok {
+			value := os.Getenv(envVar)
+			if len(value) == 0 {
 				fmt.Printf("%s is not set\n", envVar)
 				return false
 			}
 		}
 
-		re := regexp.MustCompile(`[a-z]+`)
-		scm := re.FindAllString(os.Getenv("GITOPS_REPO_URL"), 2)[1]
-
-		switch scm {
-		case "github":
-			os.Setenv("GITHUB_TOKEN", os.Getenv("GIT_ACCESS_TOKEN"))
-		case "gitlab":
-			os.Setenv("GITLAB_TOKEN", os.Getenv("GIT_ACCESS_TOKEN"))
-		default:
-			fmt.Println("SCM is not supported")
-		}
 	} else {
```

Comment:
As we have introduced variables.json now, is it possible to use variables.json for both CI and running it locally? Or we need to keep them separate?